File: app/src/main/java/com/cdtekk/product_scanner/MainActivity.java
package com.cdtekk.product_scanner;

import android.os.Bundle;
import android.util.Pair;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements DataChangeResponse {

    private ArrayList<Pair<String, String>> products = new ArrayList<>();
    private float tAmount = 0;
    private DataChangeResponse dataChangeResponse;
    private TextView textViewTotalAmount;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.buttonAddItem).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // TODO: show add item dialog box
                CustomAddItemDialog customAddItemDialog = new CustomAddItemDialog(MainActivity.this);
                customAddItemDialog.show();
            }
        });
    }

    private void AddProduct(Pair<String, String> item){
        products.add(item);

        UpdateRecyclerView();
    }

    private void UpdateRecyclerView(){
        RecyclerView recyclerView = findViewById(R.id.recyclerview);
        RecyclerViewAdapter recyclerViewAdapter = new RecyclerViewAdapter(this, products);
        recyclerViewAdapter.delegate = this;
        recyclerView.setAdapter(recyclerViewAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
    }

    @Override
    public void onDataChange(float amountUpdate){
        Toast.makeText(this, Float.toString(amountUpdate), Toast.LENGTH_SHORT).show();
        textViewTotalAmount = findViewById(R.id.textViewTotalAmount);
        textViewTotalAmount.setText(String.format("₱%.2f", amountUpdate));
    }
}

File: app/src/main/java/com/cdtekk/groceryapp/View/MainActivity.kt
package com.cdtekk.groceryapp.View

import android.app.Dialog
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.Fragment
import com.cdtekk.groceryapp.Interface.IBackPress
import com.cdtekk.groceryapp.R
import kotlinx.android.synthetic.main.dialogbox_exit.view.*
import kotlinx.android.synthetic.main.dialogbox_test.*
import kotlin.math.log

class MainActivity : AppCompatActivity(), ProductListFragment.OnFragmentChangeListener {

    var currentLoadedFrag : Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
    }

    override fun onFragmentChanged(fragmentId : Int) {
        Log.d("GROCERYAPP", fragmentId.toString())

        currentLoadedFrag = fragmentId
    }

    override fun onBackPressed() {
        if(currentLoadedFrag == R.id.productlistFragment){
            showDialog()
        } else {
            super.onBackPressed()
        }

    }

    private fun showDialog(){
        val mDialogView = LayoutInflater.from(this).inflate(R.layout.dialogbox_exit, null)

        val mBuilder = AlertDialog.Builder(this)
            .setView(mDialogView)

        val mAlertDialog = mBuilder.show()

        mDialogView.btnNo.setOnClickListener{
            mAlertDialog.dismiss()
        }

        mDialogView.btnYes.setOnClickListener{
            super.onBackPressed()
        }
    }
}
